overlord/servicestate/servicemgr_test.go

```diff
@@ -224,7 +224,9 @@ var _ = Suite(&ensureSnapServiceSuite{})
 
 func (s *baseServiceMgrTestSuite) mockSystemctlCalls(c *C, expCalls []expectedSystemctl) (restore func()) {
 	systemctlCalls := 0
+	allSystemctlCalls := [][]string{}
 	r := systemd.MockSystemctl(func(args ...string) ([]byte, error) {
+		allSystemctlCalls = append(allSystemctlCalls, args)
```

Comment:
I guess with this we could simply and remove the second "systemctlCalls++" (which looks a bit out of place to me, i.e. the fact that it's needed is only clear if you continue reading and piece together that the restore will also use this.

Either:
```
diff --git a/overlord/servicestate/servicemgr_test.go b/overlord/servicestate/servicemgr_test.go
index 0a952afff6..c9c4a43488 100644
--- a/overlord/servicestate/servicemgr_test.go
+++ b/overlord/servicestate/servicemgr_test.go
@@ -233,7 +233,6 @@ func (s *baseServiceMgrTestSuite) mockSystemctlCalls(c *C, expCalls []expectedSy
                        systemctlCalls++
                        return []byte(res.output), res.err
                }
-               systemctlCalls++
                c.Errorf("unexpected and unhandled systemctl command: %+v", args)
                return nil, fmt.Errorf("broken test")
        })
@@ -243,7 +242,7 @@ func (s *baseServiceMgrTestSuite) mockSystemctlCalls(c *C, expCalls []expectedSy
                // double-check at the end of the test that we got as many systemctl calls
                // as were mocked and that we didn't get less, then re-set it for the next
                // test
-               c.Assert(systemctlCalls, Equals, len(expCalls), Commentf("got: %v", allSystemctlCalls))
+               c.Assert(len(allSystemctlCalls), Equals, len(expCalls), Commentf("got: %v", allSystemctlCalls))
        }
 }
```
or maybe even
```
iff --git a/overlord/servicestate/servicemgr_test.go b/overlord/servicestate/servicemgr_test.go
index 0a952afff6..43f42f689c 100644
--- a/overlord/servicestate/servicemgr_test.go
+++ b/overlord/servicestate/servicemgr_test.go
@@ -223,17 +223,15 @@ After=usr-lib-snapd.mount
 var _ = Suite(&ensureSnapServiceSuite{})
 
 func (s *baseServiceMgrTestSuite) mockSystemctlCalls(c *C, expCalls []expectedSystemctl) (restore func()) {
-       systemctlCalls := 0
        allSystemctlCalls := [][]string{}
        r := systemd.MockSystemctl(func(args ...string) ([]byte, error) {
+               systemctlCalls := len(allSystemctlCalls)
                allSystemctlCalls = append(allSystemctlCalls, args)
                if systemctlCalls < len(expCalls) {
                        res := expCalls[systemctlCalls]
                        c.Check(args, DeepEquals, res.expArgs)
-                       systemctlCalls++
                        return []byte(res.output), res.err
                }
-               systemctlCalls++
                c.Errorf("unexpected and unhandled systemctl command: %+v", args)
                return nil, fmt.Errorf("broken test")
        })
@@ -243,7 +241,7 @@ func (s *baseServiceMgrTestSuite) mockSystemctlCalls(c *C, expCalls []expectedSy
                // double-check at the end of the test that we got as many systemctl calls
                // as were mocked and that we didn't get less, then re-set it for the next
                // test
-               c.Assert(systemctlCalls, Equals, len(expCalls), Commentf("got: %v", allSystemctlCalls))
+               c.Assert(len(allSystemctlCalls), Equals, len(expCalls), Commentf("got: %v", allSystemctlCalls))
        }
 }
 
```
? Anyway, just an idea :)

Reply:
I went with a slight mix of suggestions here